### PR TITLE
[NeoML] exponentFunc in CLogRegression 

### PR DESCRIPTION
**Pull request by favorart**

### CODE REVIEW

File: NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp [hunk 1]
```diff
@@ -98,13 +97,8 @@ void CTrustRegionNewtonOptimizer::Optimize()
...
 		else if (actualReduction < eta2 * predictedReduction)
 			trustRegionSize = max(sigma1 * trustRegionSize, min(alpha, sigma3 * trustRegionSize));
-		else {
-			if( reach_boundary ) {
-				trustRegionSize = sigma3 * trustRegionSize;
-			} else {
-				trustRegionSize = max( trustRegionSize, min( alpha, sigma3 * trustRegionSize ) );
-			}
-		}
+		else
```
Comment (AndrewAndrianov) [inline NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp]: missed braces in if-else blocks

File: NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp [hunk 2]
```diff
@@ -159,8 +153,8 @@ void CTrustRegionNewtonOptimizer::Optimize()
 // The solution is represented as decomposition into conjugate directions:
 // shift = sum(alpha * conjugateVector)
 
-int CTrustRegionNewtonOptimizer::conjugateGradientSearch( double trustRegionSize, 
-	const CFloatVector& gradient, CFloatVector& shift, CFloatVector& residue, bool& reach_boundary )
+int CTrustRegionNewtonOptimizer::conjugateGradientSearch(double trustRegionSize,
```
Comment (AndrewAndrianov) [inline NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp]: missed spaces in parentheses

**Review by AndrewAndrianov — DISMISSED** [review]

File: NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp (hunk 2 above)
Comment (favorart) [reply NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp]: fixed

File: NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp (hunk 1 above)
Comment (favorart) [reply NeoML/src/TraditionalML/TrustRegionNewtonOptimizer.cpp]: fixed

**Review by AndrewAndrianov — APPROVED** [review]

**Pull request merged**